### DISCUSSION TITLE
Hardware related question

### DISCUSSION THREAD

**Discussion by DexasElectronics** [Expatria-Technologies/RT_Jog_Controller#12] [Q&A] [answered]
Hello. I have a question to the person who designed this hardware. What is the purpose of 555 timer next to each button? My initial thought was maybe noise immunity / debouncing, but wouldnt a standart schmitt trigger do the job ?

**Comment by andrewmarles** [comment, marked as answer]
It is a hardware debounce/timing circuit that asserts a minimum pulse when the button is pressed. It makes the software easier since you can poll the buttons deterministically and know you are not missing presses.  A standard schmitt trigger will still bounce on a physical button (they literally bounce from on to off when pressed and released so you need hysteresis in the time domain, not just on the voltage level).

It is just the approach I took on the first joggers and so it has stuck.  You can push the same functions up and do them in software which is certainly possible with the RP2040.

**Reply by DexasElectronics** [reply]
There are schmitt triggers with histherysis, and also normally for debounce we use caps. But yeah, if you needed specific minimum pulse length, this makes sense. Was just interested in general, since simulations of schematics shown that circuit pretty much works as inverter with debounce, but didnt check the pulse length. Now it makes more sense :)
